feat: Add config to always accept the TOS on order confirmation

aragon999 opened the pull request:
### 1. Why is this change necessary?
See the related details here (German): https://feedback.shopware.com/forums/942607-shopware-6-product-feedback-ideas/suggestions/50063418-checkbox-f%C3%BCr-agb-entfernen-weil-rechtlich-unn%C3%B6tig
And here (also German): https://www.heise.de/hintergrund/Warum-im-Netz-Checkboxen-fuer-AGB-und-Datenschutz-meist-ueberfluessig-sind-10449542.html

### 2. What does this change do, exactly?
Add a configuration to show the checkbox for the TOS or always accept them.

### 3. Describe each step to reproduce the issue or behaviour.
\-

### 4. Please link to the relevant issues (if any).
\-

### 5. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/shopware/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfilled them

aragon999 committed:
feat: Add config to always accept the TOS on order confirmation

FlorianKe commented:
Wouldn’t it make more sense to remove the function entirely instead of just deactivating it?

cc @shopware/product-cc-checkout

aragon999 commented:
That could be a breaking change, but I would be fine with that, not sure about regulations outside of the EU?

Let me know, if I should change the PR to remove the TOS check entirely.

FlorianKe commented:
Good morning @aragon999,
We have decided to stick with the toggle to activate and deactivate the TOS. But, we would like to move the label 'By placing your order, you accept our...' from the top of the page to the bottom, between the summary card and the 'Submit order' button.

aragon999 commented:
@FlorianKe okay, very good. Yes let me know how it should look like :-)

FlorianKe commented:
It would be worth its weight in gold if you could implement it as shown in the image 💙  

The distance between the elements is 1rem.


<img width="406" height="465" alt="image" src="https://github.com/user-attachments/assets/7feef5dc-a725-4f3b-96b5-ce70e5dc44dd" />

aragon999 commented:
@FlorianKe okay, should we also move it, in a similar way, when we have a checkbox?

Should the snippets be replaced as well?

FlorianKe commented:
> okay, should we also move it, in a similar way, when we have a checkbox?

The plan for the next major update is to deprecate the checkbox at the top and move it to the position of your new label. Either the new label or the checkbox will be displayed at that position. If you'd like to move the checkbox as well, that would be really helpful — but it's not required. I can handle that part afterwards.

aragon999 committed:
fixup: Combine with cancelation policy and move to the bottom

aragon999 commented:
@FlorianKe okay thank you, I now refactored the PR
<img width="2792" height="3654" alt="image" src="https://github.com/user-attachments/assets/33a395a3-a8af-4a8d-956f-ebc2665a7a6b" />

With this change I also moved the "cancellation policy" into the same snippet, as I think there is no separate text required?

In my option this cleans up the checkout quite nicely.

However there is one last missing piece, which is not yet implemented, this is the "revocation" checkbox when the cart contains a digital download product.

Do you have an idea how and where to put it? Maybe below the product table? Then this checkbox is closer to the confirm button/form? And the customer is not so confused what to check?

Furthermore the question is, if we should move the placement with this new option, or with the feature flag `v6.8.0.0`?

FlorianKe commented:
> With this change I also moved the "cancellation policy" into the same snippet, as I think there is no separate text required?

That's right. Looks good!

> However there is one last missing piece, which is not yet implemented, this is the "revocation" checkbox when the cart contains a digital download product.

It would be great if you could add the section shown in the picture. Although it would take a little more time, the checkout would look even better.

<img width="947" height="642" alt="image" src="https://github.com/user-attachments/assets/13426e04-9411-477a-9e2a-77e6beaf7fa9" />

> Furthermore the question is, if we should move the placement with this new option, or with the feature flag v6.8.0.0?

As the storefront is changing quite a bit, it should be hidden behind feature flag v6.8.0.0.


Thanks for all your work!
